Unable to use dotnet client on m1 mac

To recreate (on an m1 mac), make a new directory, then inside that directory run:

```
dotnet new console
dotnet add package tigerbeetle
```

Edit Program.cs and set it to:

```
using System;

using TigerBeetle;

// Validate import works.
var id = new TigerBeetle.UInt128(1);
Console.WriteLine("SUCCESS");
```

And run `dotnet run`.

You should see this:

```
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'TigerBeetle, Version=0.13.36.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

File name: 'TigerBeetle, Version=0.13.36.0, Culture=neutral, PublicKeyToken=null'
```

But that should not happen. The script should succeed.

That's the problem, it says we published our dll for AMD64, instead of cross-platform MSIL. Needs more investigation.

```
warning MSB3270:
There was a mismatch between the processor architecture of the project being built "MSIL"
and the processor architecture of the reference ".../TigerBeetle.dll", "AMD64".
This mismatch may cause runtime failures.
Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project.
```

--- a/src/clients/dotnet/TigerBeetle.Benchmarks/TigerBeetle.Benchmarks.csproj
+++ b/src/clients/dotnet/TigerBeetle.Benchmarks/TigerBeetle.Benchmarks.csproj
@@ -3,7 +3,7 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>net6.0</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <UseCurrentRuntimeIdentifier>true</UseCurrentRuntimeIdentifier>
+    <UseCurrentRuntimeIdentifier>false</UseCurrentRuntimeIdentifier>
     <RollForward>LatestMajor</RollForward>
   </PropertyGroup>
   <ItemGroup>

--- a/src/clients/dotnet/TigerBeetle/TigerBeetle.csproj
+++ b/src/clients/dotnet/TigerBeetle/TigerBeetle.csproj
@@ -2,11 +2,12 @@
   <Import Project="TigerBeetle.props" />
   <PropertyGroup>
     <TargetFrameworks>netstandard2.1;net6.0</TargetFrameworks>
+    <Platforms>AnyCPU</Platforms>
+    <UseCurrentRuntimeIdentifier>false</UseCurrentRuntimeIdentifier>
     <LangVersion>10</LangVersion>
     <Nullable>enable</Nullable>
     <AssemblyName>TigerBeetle</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <UseCurrentRuntimeIdentifier>true</UseCurrentRuntimeIdentifier>
     <RollForward>LatestMajor</RollForward>
   </PropertyGroup>
   <PropertyGroup>

